server/sandbox.go

```diff
@@ -2,13 +2,30 @@ package server
 
 import (
 	"fmt"
+	"sync"
 
 	"github.com/docker/docker/pkg/stringid"
 	"github.com/kubernetes-incubator/cri-o/oci"
+	"github.com/containernetworking/cni/pkg/ns"
 	"k8s.io/kubernetes/pkg/fields"
 	pb "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
 )
 
+type sandboxNetNs struct {
+	sync.Mutex
+	ns     ns.NetNS
+	closed bool
```

Comment:
When ocid tries to restore a sandbox, we have 2 possibilities:

1. The sandbox has been stopped and the netns closed.
2. The sandbox is still READY and the netns is still opened.

I understand your concern is related to 1, and this is how we handle it with this PR: When trying to get the netns for a restored sandbox we set the sandbox netns to nil if it was stopped before stopping ocid:

```go
        // We add a netNS only if we can load a permanent one.
	// Otherwise, the sandbox will live in the host namespace.
	netNsPath, err := configNetNsPath(m)
	if err == nil {
		netNS, nsErr := netNsGet(netNsPath)
		// If we can't load the networking namespace
		// because it's closed, we just set the sb netns
		// pointer to nil. Otherwise we return an error.
		if nsErr != nil && nsErr != errSandboxClosedNetNS {
			return nsErr
		}

		sb.netns = netNS
	}
```

Having a close netns is actually equivalent to not having one. Please let me know if that makes sense to you.